### PR TITLE
feat: making negatives and decimals allowed on coin admin block with chips showing negative coins colored yellow

### PR DESCRIPTION
**Pull request by Logos50607**
Making negatives and decimals allowed on coin admin block with chips showing negative coins colored yellow

### CODE REVIEW

**Comment by Logos50607** [comment]
[(1)【代幣管理】可以發放負數代幣與值為小數點的代幣](https://trello.com/c/ZAk4tMiz/5299-1%E3%80%90%E4%BB%A3%E5%B9%A3%E7%AE%A1%E7%90%86%E3%80%91%E5%8F%AF%E4%BB%A5%E7%99%BC%E6%94%BE%E8%B2%A0%E6%95%B8%E4%BB%A3%E5%B9%A3%E8%88%87%E5%80%BC%E7%82%BA%E5%B0%8F%E6%95%B8%E9%BB%9E%E7%9A%84%E4%BB%A3%E5%B9%A3)

File: src/components/coin/MemberCoinAdminBlock.tsx [hunk 1]
```diff
@@ -506,10 +507,10 @@ const useCoinLogCollection = (filter?: { nameAndEmail?: string; title?: string;
       : undefined,
     member: filter?.nameAndEmail
       ? {
-          _or: [{ name: { _like: `%${filter.nameAndEmail}%` } }, { email: { _like: `%${filter.nameAndEmail}%` } }],
+          _or: [{ name: { _like: `% ${filter.nameAndEmail}% ` } }, { email: { _like: ` % ${filter.nameAndEmail}% ` } }],
```
Comment (Ponchimeow) [inline src/components/coin/MemberCoinAdminBlock.tsx:510]: Spaces can cause the LIKE operator to produce different results.

File: src/components/coin/MemberCoinAdminBlock.tsx [hunk 2]
```diff
@@ -700,20 +701,20 @@ const useOrderLogWithCoinsCollection = (filter?: {
...
         }
       : undefined,
     order_discounts: filter?.title
       ? {
-          name: { _like: `%${filter.title}%` },
+          name: { _like: `% ${filter.title}% ` },
```
Comment (Ponchimeow) [inline src/components/coin/MemberCoinAdminBlock.tsx:717]: same as above

File: src/components/common/CoinSendingModal.tsx [hunk 3]
```diff
@@ -178,9 +177,9 @@ const CoinSendingModal: React.FC<{
         </Form.Item>
         <Form.Item label={formatMessage(messages.increaseCoins)} name="amount">
           <InputNumber
-            min={1}
-            formatter={value => (parseInt(`${value}`) >= 0 ? `+${value}` : `${value}`)}
-            parser={value => value?.replace(/\D/g, '') || ''}
+            // min={1}
+            formatter={value => (Number(`${value}`) > 0 ? `+${value}` : `${value}`)}
+            // parser={value => value?.replace(/\D/g, '') || ''}
```
Comment (Ponchimeow) [inline src/components/common/CoinSendingModal.tsx:182]: could remove unused code

File: src/components/coin/MemberCoinAdminBlock.tsx (hunk 1 above)
Comment (Logos50607) [reply src/components/coin/MemberCoinAdminBlock.tsx:510]: Oops I did not touch this part and have no idea why it turns out like that. I will definitely amend it! Thanks!

File: src/components/common/CoinSendingModal.tsx (hunk 3 above)
Comment (Logos50607) [reply src/components/common/CoinSendingModal.tsx:182]: I was wondering if one day this change might be reverted though, but I will amend it as what you suggest. Thanks!

**Review by Ponchimeow — APPROVED** [review]

**Pull request merged**